### PR TITLE
[Modular] Support Ideogram4 Image2Image and Inpaint

### PR DESCRIPTION
**Pull request by lucasruan1618**
# What does this PR do?

This PR adds **img2img** and **inpainting** support for **Ideogram 4** through the Diffusers modular pipeline. The pipeline automatically selects the appropriate workflow from the supplied inputs:

- Passing `image` runs img2img.
- Passing both `image` and `mask_image` runs inpainting.

## Img2img example

```python
import torch

from diffusers import Ideogram4PromptEnhancerHead, ModularPipeline
from diffusers.utils import load_image


MODEL_ID = "ideogram-ai/ideogram-4-nf4-diffusers"
PROMPT_ENHANCER_ID = "diffusers/qwen3-vl-8b-instruct-lm-head"

pipe = ModularPipeline.from_pretrained(MODEL_ID)
pipe.load_components(dtype=torch.bfloat16)
pipe.update_components(
    prompt_enhancer_head=Ideogram4PromptEnhancerHead.from_pretrained(
        PROMPT_ENHANCER_ID, dtype=torch.bfloat16
    )
)
pipe.to("cuda")

init_image = load_image(
    "https://github.com/lucasruan1618/Image_storage/blob/main/Input/cute_cat.png?raw=true"
).convert("RGB")

prompt = "wizard dog, Gandalf-inspired, Lord of the Rings aesthetic, majestic yet cute, Studio Ghibli style"

for strength in [0.6, 0.7, 0.8, 0.9, 1.0]:
    image = pipe(
        prompt=prompt,
        image=init_image,
        height=init_image.height,
        width=init_image.width,
        strength=strength,
        prompt_upsampling=True,
        generator=torch.Generator(device="cuda").manual_seed(42),
        output="images",
    )[0]
    image.save(f"ideogram4_modular_img2img_{strength}.png")
```

## Inpainting example

```python
import torch

from diffusers import Ideogram4PromptEnhancerHead, ModularPipeline
from diffusers.utils import load_image


MODEL_ID = "ideogram-ai/ideogram-4-nf4-diffusers"
PROMPT_ENHANCER_ID = "diffusers/qwen3-vl-8b-instruct-lm-head"

pipe = ModularPipeline.from_pretrained(MODEL_ID)
pipe.load_components(dtype=torch.bfloat16)
pipe.update_components(
    prompt_enhancer_head=Ideogram4PromptEnhancerHead.from_pretrained(
        PROMPT_ENHANCER_ID, dtype=torch.bfloat16
    )
)
pipe.to("cuda")

init_image = load_image(
    "https://github.com/lucasruan1618/Image_storage/blob/main/Input/cute_cat.png?raw=true"
).convert("RGB")
mask_image = load_image(
    "https://github.com/lucasruan1618/Image_storage/blob/main/Input/mask_cat.png?raw=true"
).convert("L")

prompt = "cat wizard with red hat, gandalf, lord of the rings, detailed, fantasy, cute, adorable, Pixar, Disney"

for strength in [0.6, 0.7, 0.8, 0.9, 1.0]:
    image = pipe(
        prompt=prompt,
        image=init_image,
        mask_image=mask_image,
        height=init_image.height,
        width=init_image.width,
        strength=strength,
        prompt_upsampling=True,
        generator=torch.Generator(device="cuda").manual_seed(42),
        output="images",
    )[0]
    image.save(f"ideogram4_modular_inpaint_{strength}.png")
```

## Img2img results at different strengths

<table>
  <tr>
    <td width="100%"><b>Init image</b><br/>
      
    </td>
  </tr>
</table>

<table>
  <tr>
    <td><b>0.6</b><br/></td>
    <td><b>0.7</b><br/></td>
    <td><b>0.8</b><br/></td>
    <td><b>0.9</b><br/></td>
    <td><b>1.0</b><br/></td>
  </tr>
</table>

## Inpainting results at different strengths

<table>
  <tr>
    <td width="50%"><b>Init image</b><br/>
      
    </td>
    <td width="50%"><b>Mask image</b><br/>
      
    </td>
  </tr>
</table>

<table>
  <tr>
    <td><b>0.6</b><br/></td>
    <td><b>0.7</b><br/></td>
    <td><b>0.8</b><br/></td>
    <td><b>0.9</b><br/></td>
    <td><b>1.0</b><br/></td>
  </tr>
</table>

## Known limitation

Ideogram 4 inpainting at `strength=1.0` may produce a visible artifact in or around the masked region. At this strength, the input is fully noised and all denoising steps are used, so less information from the original masked content is preserved.

## Before submitting
- [x] Did you use an AI agent (Claude Code, Codex, Cursor, etc.) to help with this PR? If so:
  - [x] Did you read the [Coding with AI agents](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution#coding-with-ai-agents) guide?
  - [x] Did you run the [`self-review`](https://github.com/huggingface/diffusers/blob/main/.ai/skills/self-review/SKILL.md) skill on the diff?
  - [x] Did you share the final self-review notes in the PR description or a comment?
- [x] Did you read the [contributor guideline](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution)?
- [x] Did you read our [philosophy doc](https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy)? (important for complex PRs)
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?
- [x] Are you the author (or part of the team) of the model/pipeline (only applicable for model/pipeline related PRs)?

## Who can review?

- Pipelines and models: @dg845 and @asomoza 

-->

### CODE REVIEW

**Comment by lucasruan1618** [comment]
Hi @dg845, @asomoza, would you mind taking a look at this PR when you have a chance? It adds modular pipeline support for Ideogram 4 image-to-image and inpainting. Any feedback would be greatly appreciated. Thanks!
